### PR TITLE
[HB-3942] Toggle success/failure

### PR DESCRIPTION
**Pull request by chauduyphanvu**

### CODE REVIEW

File: ReferenceAdapter/src/main/java/com/chartboost/mediation/referenceadapter/sdk/ReferenceSettings.kt [hunk 1]
```diff
@@ -0,0 +1,60 @@
+package com.chartboost.mediation.referenceadapter.sdk
+
+import android.content.Context
+import java.lang.ref.WeakReference
+
+class ReferenceSettings private constructor(context: Context) {
```
Comment (bichenwang) [inline ReferenceAdapter/src/main/java/com/chartboost/mediation/referenceadapter/sdk/ReferenceSettings.kt:6]: why not just make this an object? and have get/set take context? should make this simpler to understand and not have to use synchronized.

File: ReferenceAdapter/src/main/java/com/chartboost/mediation/referenceadapter/sdk/ReferenceSettings.kt [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+        private var INSTANCE: ReferenceSettings? = null
+
+        fun getInstance(context: Context): ReferenceSettings {
+            return INSTANCE ?: synchronized(this) {
+                INSTANCE ?: ReferenceSettings(context.applicationContext).also { INSTANCE = it }
+            }
+        }
+    }
+
+    private val weakContext = WeakReference(context)
```
Comment (bichenwang) [inline ReferenceAdapter/src/main/java/com/chartboost/mediation/referenceadapter/sdk/ReferenceSettings.kt:25]: and you don't have to do something like this. storing app context is always okay. so just store `private val context = context.applicationContext` if you need it

Comment (chauduyphanvu) [reply ReferenceAdapter/src/main/java/com/chartboost/mediation/referenceadapter/sdk/ReferenceSettings.kt]: Good call. I wasn't thinking of application Context. New commit up to simplify all this.

File: ReferenceAdapter/src/main/java/com/chartboost/mediation/referenceadapter/sdk/ReferenceSettings.kt [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+ * Use of this source code is governed by an MIT-style
+ * license that can be found in the LICENSE file.
+ */
+
+package com.chartboost.mediation.referenceadapter.sdk
+
+import android.content.Context
+
+object ReferenceSettings {
+    lateinit var appContext: Context
```
[the remark this answers is not in the document]
Comment (chauduyphanvu) [reply ReferenceAdapter/src/main/java/com/chartboost/mediation/referenceadapter/sdk/ReferenceSettings.kt:13]: It's set from Canary: https://github.com/ChartBoost/android-helium-sdk/blob/develop/HeliumCanary/src/main/java/com/chartboost/heliumcanary/HeliumCanaryActivity.kt#L95. 

But yeah there's an edge case where the Canary settings toggle for "auto init" is turned off (i.e. switching to manual init) then this context will be null, and when someone tries to toggle the Reference Adapter settings they won't get saved to the SharedPrefs. 

I'll make this optional.

**Review by bichenwang — APPROVED** [review]

**Comment by bichenwang** [comment]
oh also don't forget to update the version number (I'll leave it up to you if you want to do 1.1.0 or 1.0.2) and publish it when you're done with this

**Review by cb-jpadilla — APPROVED** [review]
✔️

**Review by CB-RyanMcCormick — APPROVED** [review]

**Pull request merged**